pig/src/test/java/org/jboss/pnc/bacon/pig/impl/addons/scanservice/ValidationTest.java

```diff
@@ -0,0 +1,167 @@
+        }
+    }
+
+    @Test
+    void negativeKeywordsTest() {
+        HashMap<String, Object> shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("productID");
+        shallowCopy.put("product-id", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("eventId");
+        shallowCopy.put("event-id", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("isManagedService");
+        shallowCopy.put("is-managed-service", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("cpaasVersion");
+        shallowCopy.put("CpaasVersion", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("jobUrl");
+        shallowCopy.put("xyz", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceUrl");
+        shallowCopy.put("service-url", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceSecretKey");
+        shallowCopy.put("service-key", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceSecretValue");
+        shallowCopy.put("service-val", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("brewBuilds");
+        shallowCopy.put("brew_builds", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("scmUrl");
+        shallowCopy.put("SCMURL", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("scmRevision");
+        shallowCopy.put("scm-revision", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("extraScmUrls");
+        shallowCopy.put("xtraScmUrls", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("extraBuilds");
+        shallowCopy.put("extraBuild_", null);
+        negativeTest(shallowCopy);
+    }
+
+    void negativeTest(HashMap<String, Object> testMap) {
+        try {
+            scanService.validateMapKeys(testMap);
+            Assertions.fail();
+        } catch (Exception e) {
+            Assertions.assertTrue(true);
```

Comment:
This try-catch can have only two outcomes. Since validateMapKeys call throws an exception when validation fails we know apriori which case is success and which one is failure. We do not use try-with-resource since no resources are acquired for this validation call.

Reply:
Maybe you just wanted to use `assertThrows()` here if the test passes only when a certain exception is thrown.

Reply:
Since `assertTrue(true)` is a no-op, it does not make things more clear (to me).

If you require a certain exception to be thrown, you would use `assertThrows()`. If you require a certain exception to not be thrown, you would use `assertDoesNotThrow()`.

If you require that no exceptions are thrown, then you don't need to do anything since any exception will automatically fail the test.

Reply:
Done